boogie-org/boogie
# Allow keeping SMT commands unchanged when Boogie identifiers are renamed

Related Dafny issue: https://github.com/dafny-lang/dafny/issues/1585

This PR includes changes https://github.com/boogie-org/boogie/pull/453

SMT solvers in practice can show significantly different behavior in terms of total runtime and `check-sat` result when identifiers in the input are renamed. Such changes in solver output are often surprising to the Boogie programmer, so this PR introduces an option to normalize names in the Boogie program.

### Changes
- Enables normalizing any names that occur in the Boogie program when translating it into SMT commands.
- Remove unused code that was in SMTLibNamer and UniqueNamer

### Testing
- Add a test that checks that renaming various elements in a Boogie program does not effect the solver input.
- Add the `/normalizeNames:1` option to all tests that inspect the SMT model, to verify that names in the parsed SMT model are correctly translated back to their original name.
